xdp/src/netlink.rs

```diff
@@ -21,6 +24,51 @@ use {
 
 const NLA_HDR_LEN: usize = align_to(mem::size_of::<nlattr>(), NLA_ALIGNTO as usize);
 
+#[repr(C)]
+#[allow(non_camel_case_types)]
+pub(crate) struct ifinfomsg {
+    ifi_family: u8,
+    ifi_type: u16,
+    ifi_index: u32,
+    ifi_flags: u32,
+    ifi_change: u32,
+}
+
+#[inline]
+fn is_supported_route_type(ty: u8) -> bool {
+    matches!(
+        ty,
+        RTN_UNICAST | RTN_LOCAL | RTN_BROADCAST | RTN_MULTICAST | RTN_BLACKHOLE | RTN_THROW
+    )
+}
+
+#[inline]
+fn is_supported_route_table_id_u8(table: u8) -> bool {
+    table == RT_TABLE_UNSPEC || table == RT_TABLE_MAIN || table == RT_TABLE_LOCAL
+}
+
+#[inline]
+fn is_supported_route_table_id_opt_u32(table: Option<u32>) -> bool {
```

Comment:
Do we need 2 nearly identical functions (compare with is_supported_route_table_id_u8)? Maybe the option can be handled with unwrap_or(true) at the call site?